File: SEMESTRUL 5/Mobile Application Programming/Laboratories/Laboratory 2/app/src/main/java/com/example/tripmate/model/Trip.kt
package com.example.tripmate.model

import java.time.LocalDate
import java.util.UUID

data class Trip(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val destination: String,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val totalBudget: Double,
    val notes: String?
)

File: SEMESTRUL 5/Mobile Application Programming/Laboratories/Laboratory 3/lib/models/trip.dart
class Trip {
  final String id;
  String name;
  String destination;
  DateTime startDate;
  DateTime endDate;
  double totalBudget;
  String notes;

  Trip({
    required this.id,
    required this.name,
    required this.destination,
    required this.startDate,
    required this.endDate,
    required this.totalBudget,
    required this.notes,
  });

  Trip copyWith({
    String? name,
    String? destination,
    DateTime? startDate,
    DateTime? endDate,
    double? totalBudget,
    String? notes,
  }) {
    return Trip(
      id: id,
      name: name ?? this.name,
      destination: destination ?? this.destination,
      startDate: startDate ?? this.startDate,
      endDate: endDate ?? this.endDate,
      totalBudget: totalBudget ?? this.totalBudget,
      notes: notes ?? this.notes,
    );
  }
}
